lib/iris/common/resolve.py

```diff
@@ -1351,6 +1546,7 @@ def _prepare_points_and_bounds(
 
     @property
     def _src_cube(self):
+        assert self.map_rhs_to_lhs is not None
```

Comment:
Defensive code to ensure the `_init` restructuring to `__inti__` et al, hasn't had any side-effects that we don't know about.

But in general, it a healthy `assert` to adopt